src/main/scala/com/fulcrumgenomics/fasta/CollectAlternateContigNames.scala

```diff
@@ -213,7 +216,18 @@ class CollectAlternateContigNames
 
     // Apply to an existing sequence dictionary if necessary
     val dict: SequenceDictionary = existing match {
-      case None => SequenceDictionary(metadatas.toSeq:_*)
+      case None       =>
+        if (!sortBySequencingRole) SequenceDictionary(metadatas.toSeq:_*) else {
+          // Use the user-provided roles, otherwise use all role
+          val roles = (if (sequenceRoles.nonEmpty) sequenceRoles.toIndexedSeq else SequenceRole.values)
+            .zipWithIndex
+            .map { case (role, index) =>  role.key -> index }
+            .toMap
+          val metas = metadatas.toSeq.sortBy { metadata =>
+            metadata.get(Column.SequenceRole.tag).map { name => roles(name) }.getOrElse(roles.size)
```

Comment:
Should this be?

```suggestion
            metadata.get(Column.SequenceRole.tag).flatMap { name => roles.get(name) }.getOrElse(roles.size)
```

If I'm understanding the `.getOrElse(roles.size)` correctly, what that is doing is for when a metadata has a role that is not in the provided (or known) list of roles then the metadata entries will be forced to sort to the "end" of the metadata list?